src/framework/results/experiment_result.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_results_experiment_result_hpp_
+#define _aer_framework_results_experiment_result_hpp_
+
+#include "framework/results/experiment_data.hpp"
+
+namespace AER {
+
+//============================================================================
+// Result container for Qiskit-Aer
+//============================================================================
+
+struct ExperimentResult {
```

Comment:
This interface is very similar to `Result` one, Didn't want to inherit from it?

Reply:
`Result` contains `ExperimentResult` objects as members though

Reply:
Inherit by composition, perfect.